pkg/gcp/gcp.go

```diff
@@ -197,3 +241,33 @@ func (c *GCSClient) Close(ctx context.Context) {
 func (c *GCSClient) ObjectIsNotFound(err error) bool {
 	return errors.Is(err, gcpstorage.ErrObjectNotExist)
 }
+
+// newHTTPClient creates a new HTTP client for interacting with Google Cloud APIs.
+func newHTTPClient(ctx context.Context, o *options) (*http.Client, error) {
+	secret := o.secret
+	proxyURL := o.proxyURL
+
+	var creds *google.Credentials
+	var err error
+	if secret != nil {
+		creds, err = google.CredentialsFromJSON(ctx, secret.Data["serviceaccount"], gcpstorage.ScopeReadOnly)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Google credentials from secret: %w", err)
+		}
+	} else {
+		creds, err = google.FindDefaultCredentials(ctx, gcpstorage.ScopeReadOnly)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find Google default credentials: %w", err)
+		}
```

Comment:
For the `aws` provider using the MinIO client [we are able to use a proxy for talking to the STS endpoint](https://github.com/fluxcd/source-controller/pull/1552/files#diff-6e52b6d8d0e1e68088437c85d9ec1ec42cbc5620adf43ae661759571b43f555cR118-R138), but not here (it's not impossible but I think we would need to write or copy/adapt a significant amount of code from these Google libraries, and I'd really like to avoid that, it'd be hard to maintain). Should we document this?

Reply:
Actually this comment is more specific about not being easy to use a proxy when talking to the STS endpoing for Google. (By the way, this explicit call to `google.FindDefaultCredentials()` is no longer needed in the code, because in the latest version we just let `htransport.NewTransport()` call it internally for us).

This outdated code snippet does show that using a single `*google.Credentials` variable was handy for just passing a single `option.WithCredentials()` option to `htransport.NewTransport()` after the if-else, but that is no longer true. (But there's still a strong reason for using `google.CredentialsFromJSON()` instead of `option.WithCredentialsJSON()`, see the other thread.)